src/mlang/m_ir/mir_dependency_graph.ml

```diff
@@ -57,20 +67,36 @@ let add_usages (lvar : Mir.Variable.t) (e : Mir.expression Pos.marked) (acc : G.
   let usages = get_used_variables_ e Mir.VariableMap.empty in
```

Comment:
So here keeping only one dependency graph, in which rules are nodes, would be better. The edges in ocamlgraph can also carry labels. Here the labels would be the variables inside the rules that create dependencies. See this example https://github.com/CatalaLang/catala/blob/master/compiler/scopelang/dependency.ml for how to create labels on edges and how using them to display good error messages for cycle detection.